lib/ngx/re.lua

```diff
@@ -168,6 +152,24 @@ function _M.split(subj, regex, opts, ctx, max, res)
                 break
             end
 
+            if from == to then
+                local old_pos = ctx.pos
+
+                local from2, _, _, err2 = re_split_helper(subj, compiled,
+                                              compile_once, flags, ctx)
+                if err then
+                    return nil, err2
+                end
+
+                if not from2 then
+                    break
+                end
+
+                from = from2
+                to = from2
+                ctx.pos = old_pos
```

Comment:
Maybe we should not backtrack to the old position? This way we can make this extra match not get wasted. The same applies to the `to` index returned by this extra match.